### PR TITLE
Improve relevance of search results

### PR DESCRIPTION
**Pull request by neilbmclaughlin**
Create search words compound field and index it on application start to
improve the effectiveness of search

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-54.dev.c2s.nhschoices.net](http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-54.dev.c2s.nhschoices.net](http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-54.dev.c2s.nhschoices.net](http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by neilbmclaughlin** [comment]
The integration tests exercise the code but there are no specific ones for the changes in search. I'd like to address this as a separate bit of work using `rewire`.

File: app/views/includes/result-item.nunjucks [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <h3 class="results__name">
-  {% if gp.supplier %}
-    <a href="{{gp.supplier}}">{{ gp.name }}</a>
+  {% if gp.bookingSystem %}
+    <a href="{{gp.bookingSystem.bookOnlineLink}}">{{ gp.name }}</a>
```
Comment (st3v3nhunt) [inline app/views/includes/result-item.nunjucks]: This is optional. Suggest adding the check to the line above e.g. `{% if gp.bookingSystem && gp.bookingSystem.bookOnlineLink %}`

File: app/views/includes/result-item.nunjucks [hunk 2]
```diff
@@ -9,7 +9,7 @@
   {% if gp.address.addressLines %} {{ gp.address.addressLines | safe | replace(",", ", ") }}, {% endif %}
   {% if gp.address.postcode %} {{ gp.address.postcode }} {% endif %}
 {% endif %}
-{% if gp.supplier %}
+{% if gp.bookingSystem %}
```
Comment (st3v3nhunt) [inline app/views/includes/result-item.nunjucks]: Could a `not` be added here e.g. `{% if not gp.bookingSystem %}` to avoid the empty block?

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-54.dev.c2s.nhschoices.net](http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-54.dev.c2s.nhschoices.net](http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-54.dev.c2s.nhschoices.net](http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:warning: deployment of 54 for nhsuk/gp-finder to rancher stack gp-finder-pr-54 failed

**Comment by c2s-dev** [comment]
:warning: deployment of 54 for nhsuk/gp-finder to rancher stack gp-finder-pr-54 failed

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-54.dev.c2s.nhschoices.net](http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-54.dev.c2s.nhschoices.net](http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployed to [http://gp-finder-pr-54.dev.c2s.nhschoices.net](http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-54 in nhsuk-dev succeeded (http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-54 in nhsuk-dev succeeded (http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder-pr-54 in nhsuk-dev succeeded (http://gp-finder-pr-54.dev.c2s.nhschoices.net)

**Pull request merged**